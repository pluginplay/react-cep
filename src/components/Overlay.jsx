import Flex from 'styled-flex-component'

export default Flex.extend`
  transition: opacity 0.25s linear, z-index 0s linear ${props => props.visible ? '' : '0.25s'};
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: ${props => props.theme.background.dark};
  z-index: ${props => props.visible ? 3 : -1};
  opacity: ${props => props.visible ? 1 : 0};
  a {
    color: white;
    text-decoration: none;
    opacity: 0.75;
    transition: opacity 0.2s linear;
    &:hover {
      opacity: 1;
    }
  }
`
