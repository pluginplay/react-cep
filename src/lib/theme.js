import * as csInterface from 'cep-lib/csinterface'
import { lighten, darken } from 'polished'

export const getDefaultTheme = () => {
  const ae = csInterface.getThemeInformation()
  ae.systemHighlightColor = '#0078d7' // We have to hard-code this because Windows gets the wrong color.
  return {
    ae,
    background: {
      lighter: lighten(0.2, ae.panelBackgroundColor),
      light: lighten(0.1, ae.panelBackgroundColor),
      dark: darken(0.05, ae.panelBackgroundColor),
      darker: darken(0.1, ae.panelBackgroundColor)
    },
    colors: {
      green: '#23EB87',
      red: '#eb093f',
      blue: ae.systemHighlightColor
    }
  }
}
