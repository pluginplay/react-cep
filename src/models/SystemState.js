import { observable, action } from 'mobx'

export default class SystemState {
  @observable progress = 0
  @observable progressLabel = 'Loading'
  @observable showProgress = false
  @observable showProgressComplete = false
  @observable errorMessage = null
  @observable errorObject = null
  @observable showErrorOverlay = false
  @observable debugMessage = null
  @observable showDebugOverlay = null
  @observable confirmationMessage = null
  @observable confirmationCallback = null
  @observable showConfirmationOverlay = false

  @action setProgress (progress, label, showProgress = true) {
    this.showProgressComplete = false
    this.showProgress = showProgress
    this.progressLabel = label
    this.progress = progress
  }

  @action progressComplete () {
    this.progress = 100
    this.showProgressComplete = true
    setTimeout(() => {
      this.hideProgress()
    }, 3000)
  }

  @action hideProgress () {
    this.showProgress = false
  }

  @action setErrorMessage (errorText, error) {
    this.errorMessage = errorText
    this.errorObject = error
    this.showErrorOverlay = true
  }

  @action hideErrorMessage () {
    this.errorMessage = ''
    this.errorObject = null
    this.showErrorOverlay = false
  }

  @action addDebugMessage (message) {
    console.log(`[DEBUG] ${message}`)
    if (this.debugMessage === null) {
      this.debugMessage = message
    } else {
      this.debugMessage += `\n${message}`
    }
  }

  @action showDebug () {
    this.showDebugOverlay = true
  }

  @action clearDebug () {
    this.showDebugOverlay = false
    this.debugMessage = null
  }

  @action showConfirmation (message, callback) {
    this.showConfirmationOverlay = true
    this.confirmationMessage = message
    this.confirmationCallback = callback
  }

  @action hideConfirmation () {
    this.showConfirmationOverlay = false
    this.confirmationMessage = null
    this.confirmationCallback = null
  }
}
