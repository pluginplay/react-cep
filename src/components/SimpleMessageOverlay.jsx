import React from 'react'
import PropTypes from 'prop-types'
import Overlay from './Overlay'
import { observer } from 'mobx-react'
import styled from 'styled-components'

@observer
class SimpleMessageOverlay extends React.Component {
  render () {
    const visible = this.props.systemState.simpleMessageVisible
    const message = this.props.systemState.simpleMessage
    return (
      <Overlay visible={visible} className={this.props.className} center column>
        <p>{message}</p>
      </Overlay>
    )
  }
}

SimpleMessageOverlay.propTypes = {
  systemState: PropTypes.object.isRequired
}

export default styled(SimpleMessageOverlay)`
  background: none;
  transition: opacity ${props => props.visible ? '0.1s' : '0.25s'} linear, z-index 0s linear ${props => props.visible ? '' : '0.25s'};
  > p {
    padding: 10px 15px;
    background: ${props => props.theme.background.darker};
    border: solid 1px ${props => props.theme.background.light};
    box-shadow: 0 0 3px ${props => props.theme.background.dark};
    border-radius: 2px;
  }
`
