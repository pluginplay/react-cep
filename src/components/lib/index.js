import SystemState from '../../models/SystemState'
import PersistedState from '../../models/PersistedState'

export { SystemState, PersistedState }
