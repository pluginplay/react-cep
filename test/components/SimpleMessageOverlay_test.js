import React from 'react'
import { shallow } from 'enzyme'
import { expect, withTheme } from '../setup'
import renderer from 'react-test-renderer'
import SystemState from '../../src/components/SystemState'
import SimpleMessageOverlay from '../../src/components/SimpleMessageOverlay'

import Overlay from '../../src/components/Overlay'
import { getDefaultTheme } from '../../src/lib/theme'

let systemState = null

beforeEach(() => {
  systemState = new SystemState()
  systemState.simpleMessageVisible = true
  systemState.simpleMessage = 'Hello world!'
})

const getWrapper = () => {
  return shallow(<SimpleMessageOverlay theme={getDefaultTheme()} systemState={systemState} />).dive()
}

it('is only visible when simpleMessageVisible is true', () => {
  const wrapper = getWrapper()
  expect(wrapper.find(Overlay).first()).to.have.prop('visible').equal(true)
})

it('displays the proper message', () => {
  const wrapper = getWrapper()
  expect(wrapper.find('p')).to.have.text('Hello world!')
})

it('renders the proper HTML structure', () => {
  const tree = renderer
    .create(withTheme(<SimpleMessageOverlay systemState={systemState} />))
    .toJSON()
  jestExpect(tree).toMatchSnapshot()
})
