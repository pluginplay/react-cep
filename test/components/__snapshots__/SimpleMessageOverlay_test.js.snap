// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the proper HTML structure 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: opacity 0.25s linear,z-index 0s linear;
  transition: opacity 0.25s linear,z-index 0s linear;
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #191919;
  z-index: 3;
  opacity: 1;
}

.c1 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: 0.75;
  -webkit-transition: opacity 0.2s linear;
  transition: opacity 0.2s linear;
}

.c1 a:hover {
  opacity: 1;
}

.c0 {
  background: none;
  -webkit-transition: opacity 0.25s linear,z-index 0s linear 0.25s;
  transition: opacity 0.25s linear,z-index 0s linear 0.25s;
}

.c0 > p {
  padding: 10px 15px;
  background: #0c0c0c;
  border: solid 1px #404040;
  box-shadow: 0 0 3px #191919;
  border-radius: 2px;
}

<div
  className="c0 c1"
>
  <p>
    Hello world!
  </p>
</div>
`;
